pkg/network/ndmz/dualstack.go

```diff
@@ -303,6 +303,39 @@ func (d *dmzImpl) GetIP(family int) ([]net.IPNet, error) {
 	return results, err
 }
 
+// Get gateway to given destination ip
+func (d *dmzImpl) GetDefaultGateway(destination net.IP) (net.IP, error) {
+	netns, err := namespace.GetByName(dmzNamespace)
+	if err != nil {
+		return nil, err
+	}
+
+	defer netns.Close()
+
+	var gw net.IP
+	err = netns.Do(func(_ ns.NetNS) error {
+		routes, err := netlink.RouteGet(destination)
+		if err != nil {
+			return err
+		}
+
+		for _, route := range routes {
+			if route.Gw != nil {
+				gw = route.Gw
+				return nil
+			}
+		}
```

Comment:
should we check that the route returned here is the default one? or do we just need any route.

Reply:
I think the default gw is the only one that will have `gw` set actually. I tried to check the Src also to be nil but seems that the RouterGet uses your local ip as a source which makes sense

Reply:
I also use a public IP so the only one is returned is a default gw with the betst metric